validator/spanner_location.rego

```diff
@@ -0,0 +1,63 @@
+#
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package templates.gcp.GCPSpannerLocationConstraintV1
+
+import data.validator.gcp.lib as lib
+
+deny[{
+	"msg": message,
+	"details": metadata,
+}] {
+	constraint := input.constraint
+	lib.get_constraint_params(constraint, params)
+
+	asset := input.asset
+
+	# Applies to spanner instances only
+	asset.asset_type == "spanner.googleapis.com/Instance"
+
+	# Retrieve the list of allowed locations - default to empty set
+	locations := cast_set(lib.get_default(params, "locations", {}))
+
+	# The asset raises a violation if location_is_valid is evaluated to false
+not location_is_valid(	asset, locations)
```

Comment:
nit. This should be aligned. Make sure to run `make format`.

Reply:
actually make format is what broke this indentation... I have noticed that some key words get misaligned with this opa fmf command (like not and re_match for instance).

I can fix it but next times we run make build or make format it will come back.